torch/_torch_docs.py

```diff
@@ -4635,6 +4634,57 @@ def merge_dicts(*dicts):
     torch.return_types.median(values=tensor([-0.3982,  0.2270,  0.2488,  0.4742]), indices=tensor([1, 4, 4, 3]))
 """.format(**single_dim_common))
 
+add_docstr(torch.nanmedian,
+           r"""
+nanmedian(input) -> Tensor
+
+Computes the median of the values in :attr:`input`, ignoring ``NaN``s.
+
+This function is identical to :func:`torch.median` when there are no ``NaN`` values in :attr:`input`.
+When :attr:`input` has one or more ``NaN`` values, :func:`torch.median` will always return ``NaN``,
+while this function will return the median of the non-``NaN`` elements in :attr:`input`.
+If all the elements in :attr:`input` are ``NaN`` it will also return ``NaN``.
+
+Args:
+    {input}
+
+Example::
+
+    >>> a = torch.tensor([1, float('nan'), 3, 2])
+    >>> a.median()
+    tensor(nan)
+    >>> a.nanmedian()
+    tensor(2.)
+
+.. function:: nanmedian(input, dim=-1, keepdim=False, *, out=None) -> (Tensor, LongTensor)
+
+Computes the median of the values, and its index, along dimension :attr:`dim` of :attr:`input`, ignoring ``NaN``s.
```

Comment:
Should this be "Computes the medians of the values, and their indices"? The way it's currently worded suggests only a single index is returned.

Maybe there's a better way to say this?

Reply:
I reworded a bit. I also fixed the namedtensor tests.